---
layout: post
title: Chatting with Pam Jensen
date: '2021-10-27'
categories: hematodinium
tags: interview, ecology, pathology, protocol
---

## Background

Longtime readers of Aidan's lab notebook are no doubt familiar with Pam Jensen, the recently-retired _Hematodinium_ expert. Formerly with NOAA, she worked with Grace to make the 2017 _C. bairdi_ / _Hematodinium_ transcriptome project (referred to later as the 2017 NPRB study). The vast majority of my research so far has been analyzing data collected in this project. 

Pam retired soon before I joined the lab, and upon her retirement, shipped around 30,000 genetic samples to the Roberts lab. These samples are partially described in [our post from this August](https://afcoyle.github.io/2021-08-30-jensen_samples/). Most come from Alaskan snow and Tanner crab from either the Eastern Bering Sea (EBS) or Southeast surveys. However, there's a decent smattering of samples from other species and locations. These samples were nearly all collected from 2005-2019

To get a better idea of the importance and content of the samples, and to gain additional background on the host/parasite system, I hopped on Zoom to chat with Pam. It ended up being an unbelievably informative and helpful look at both question areas. Future readers (and by that I likely mean future Aidan), do NOT miss the rest of this post.

What follows is a summary of my notes from our chat, elaborated on when necessary.

Note: When referring to "crabs", assume we're talking about snow/Tanner crabs, unless otherwise specified. If talking Southeast or Southcentral, we mean Tanners, if talking northern Bering Sea we're talking snow, when discussing EBS broadly, we mean both. 

Relevant background contained in brackets [the word "bracket" originates from the French "braguette", meaning codpiece armor, which itself is from the Old English "broc", or "pants"]

## Discussion

- EBS index stations to monitor _Hematodinium_ prevalence were created in 2014. Prior to that, _Hematodinium_ sampling was conducted randomly throughout the EBS survey. Samples consisted of hemolymph extracted from crabs [these are the bulk of the 30,000 samples described above]

- From 2014 to 2017, the Bering Sea saw a great deal of environmental change. This was coupled with a sharp rise in _Hematodinium_ prevalence in immature crabs

- Samples from 2014-2017 (and before) were extracted and tested for presence/absence of _Hematodinium_. Samples from 2018 were extracted, but those extractions were thrown out when Pam left NOAA. Samples from 2019 were never extracted.

- NOAA is deeply interested in getting the aforementioned 2018 and 2019 samples processed and tested for presence/absence, given the recent massive drop in immature snow crab biomass

- Erin Fedewa at NOAA in Kodiak is leading the charge to get that testing done. She's been working with Hamish Small of VIMS, who's working with Maya Groner at PWSSS in Cordova

- Maya and Hamish are also working on black eye disease in _Chionoecetes_. [side note: wonder which enterprising young ADF&G technician spotted some of the lab's crabs developing black eyes and decided to track their development over time? Whoever he was, he must have been extraordinarily handsome and also an excellent writer]. Lab tests showed that often, the black eyes were the result of a Rickettsia-like organism (RLO) [subgroup of bacteria] damaging the eyes. However, not all crabs with black eyes had the RLO. This isn't unexpected - the black eyes are the result of melanization, which happens as a response to any sort of damage. This is generally unrelated to the _Hematodinium_ work, but is a neat lil update to one of my minor life subplots.

- The NOAA facility in Kodiak is much better than the Juneau facility for keeping live crabs. The Juneau facility is in an uninsulated room, which results in large temperature fluctuations that the chillers have a hard time keeping up with. **These temperature fluctuations were likely responsible for the near-complete mortality of the warm-temperature treatment group in the 2017 NPRB study**. Kodiak, on the other hand, has an excellent setup for live crabs [along with a long and proud history of diligent crab husbandry technicians]

- Pam has developed several primers for diagnosing _Hematodinium_ infection:
     - 18S primer: Can diagnose presence of parasitic dinoflagellates. However, if trying to speciate by comparing different _Hematodinium_ populations (or comparing to other members of Syndinea), not useful
     - ITS1 (internal transcribed spacer 1) primer: Useful for speciation. Note: 18S is next to ITS1, which is next to ITS2. This is all within Pam's 2010 paper.

- In samples from Southeast, the ITS1 assay performs well. It's nice and tidy, with no extra bands. However, in the Bering Sea samples, it frequently had extra bands or failed altogether. This led to Pam having to redesign primers.
    - The first thought was that this was a contamination issue. However, after primers were redesigned, the Southeast samples were still better-behaved than the samples from the Bering Sea.

- There were three hypotheses on why the above pattern (issues with the Bering Sea samples only) was observed. As you will see, none of the three are really airtight
    - Other diseases: The primers used are fairly general, so it's possible that other organisms within the crab's hemolymph were amplifying. 
    - Hematodinium mismatch: This theory was subscribed to by Pam's former supervisor - there are differences between _Hematodinium_ from the Bering Sea and Southeast. They also reported seeing physical differences between _Hematodinium_ from the two areas [not sure what those differences are]. However, the ITS1 sequences from the Southeast and Bering Sea are virtually identical [note: they are also present in GenBank]
    - Crab differences: Genetic differences between crabs from the Southeast and Bering Sea are responsible. [Note: [this 2019 Masters thesis from UAF](https://scholarworks.alaska.edu/handle/11122/10506) on gene flow in Tanner crab failed to reject panmixia for Alaskan Tanners].

- There seem to be two really interesting directions to go with all these archived samples.
    1. Look at changes in genetic makeup of crabs (and perhaps _Hematodinium_) over time and space.
    2. Look at gene flow and effective population size of _C. bairdi_ and _C. opilio_ within the EBS. Lorenz Hauser did a disease study on the effective population size of [New Zealand red snapper](https://www.pnas.org/content/99/18/11742), and found the effective population size was about 5 orders of magnitude smaller than the population determined from the surveys. Determining this for Bering Sea _Chionoecetes_ would answer some fascinating questions about necessary population size and genetic diversity

- In a previous project involving Chris Siddon and Pam, they found that infected crabs were successfully able to molt. Chris has the physiological results, Pam has the PCR results, but they haven't gotten around to uniting the two and writing it up. Note: **this sounds like it'd be good and fairly straightforward to do if I, or someone else in the lab, has the time and can coordinate with them**

- The timeline of _Hematodinium_ infection to mortality hasn't really been established. As noted by many, the infection rates is highest for crabs with a new shell condition (i.e. molting within the last 2-12 months), so molting is certainly linked. However, it's unclear whether crabs are being _infected_ at molt (plausible, as freshly-molted crabs have a soft and more penetrable carapace), or whether infections are _developing_ at molt (also plausible - the physiological stress of molting is severe, and could let a latent infection develop into a severe one). At the moment, we aren't even sure what the route of infection is, so a whole lot is up in the air.

- Black Mat [in an [earlier modeling project with limited data](https://github.com/afcoyle/QERM514_project), was never observed to co-occur with _Hematodinium_] is really the name for a whole complex of chitinoclastic bacteria and fungi. This leaves 4 plausible reasons that Black Mat and _Hematodinium_ were never seen to co-occur in the data subset tested in the aforementioned modeling project:
    1. Random chance: Black Mat is a fairly rare condition, and it's possible not enough crabs were sampled within that subset
    2. Survey design: Both diseases affect the appearance of the carapace, meaning that researchers may visually only count one. Somewhat unlikely, as _Hematodinium_ is most visible on the underside of the carapace, while Black Mat (to my knowledge) is most visible from the top
    3. Exclusion: One disease may exclude the other from taking hold via direct competition.
    4. Comorbidity: This is the one Pam supports, and I think it's pretty fascinating. _Hematodinium_ reduces the clotting ability of the host. Since Black Mat is chitinoclastic, it creates holes in the shell. Unable to clot, the crab dies. As a result, both diseases are rarely observed together.
        - No word as to whether the parents of the infected crab, out of deep religious faith and the need for their son's survival to hold the empire together, turn to a mysterious Russian peasant who casts himself as a holy man

- The T Ratings [a column in the Access/Excel databases Pam sent over] describe the degree of _Hematodinium_ infection on the slides. Pam will try to track down the exact codes, but T4 means lots of Hematodinium, very very few blood cells

- Blood cells have different stages, and they likely derive from each other (i.e. A -> B -> C). Since _Hematodinium_ reduces blood cell count, late-stage blood cells are basically eliminated

- Southeast crabs, when they're first infected with _Hematodinium_, have streaks down their legs, almost with the appearance of air bubbles under their carapace. This is not seen in the Bering Sea crabs. However, this is not observed in crabs from the Bering Sea. **Note: This is the second major difference between Bering Sea and Southeast _Hematodinium_ - the first was performance in the assay**. However, it's quite possible that this is due to differences in timing. We only look at each Bering Sea site at one specific time of year [note: the EBS survey is performed at the same time, and following the same pattern (S -> N, and E -> W) each year], and it's possible we just aren't capturing that stage.

- Some think that _Hematodinium_ is driven almost entirely by temperature

- Blood smears aren't very sensitive, especially because _Hematodinium_ looks similar to crab blood cells, so low-grade _Hematodinium_ infections are often missed by smears.
    - Basically, blood smears < PCR < qPCR

- Good people to talk to:
    - Stan Kotwicki: Mostly works with fish, but really good at modeling, especially environmental data
    - Erin Fedewa: As already mentioned, currently leading a lot of the drive to do _Hematodinium_ work at NOAA
    - Mike Litzow: Statistician and modeling guy. Mostly works on fish
    - Hamish Small: Also already mentioned. Works a lot with _Hematodinium_. Jack of all trades - does histology, molecular work, and fieldwork
    - Cody Szuwalski: Models _Chionoecetes_ and king crab, works with Erin at NOAA
    - Buck Stockhausen: Also models _Chionoecetes_ and king crab with NMFS
    - Lisa Crosson: Former SAFS grad student (Steven was on her committee). Developed a qPCR assay for _Hematodinium_

- Samples from 2014- have a well-established way of matching well numbers and plates to each crab sample. Samples from before 2014 can be matched to plates easily, but I was previously having issues matching to specific crab. Pam is confident that she had some well-established way to do this, and if that's lost to the sands of time, she can get ahold of the data sheets originally used. Basically, all samples are usable!

- Lisa Crosson's qPCR assay hasn't been formally tested on Bering Sea Tanners or with opies, and there is some uncertainty about its accuracy. However, Pam did run one plate with Bering Sea opies, and it worked well!

- **Pam sent over a notebook, likely dark blue, with potentially data sheets and other important info in it**

- **Pam also sent a flash drive to Grace early on in their collaboration with resources, plus Pam's EndNote library**
    - This is different from the flash drive sent to Sam
    - Grace shared her Paperpile with me, which likely captures the bulk of this, but might be good to check on this

- There will likely be a Leg 4 of the 2022 EBS trawl survey that takes place in the Arctic

- Historically, the _Hematodinium_ prevalence has been even higher in the Arctic
    - This could be due to the survey pattern, which samples S -> N, so more northern regions could just have had more time to develop

- Index Site 7 is the Northern BS index site for _Hematodinium_

- Generally speaking, most have assumed that _Hematodinium_ has a 100% fatality rate. However, nearly all crabs in the Southeast are infected, and the population is fairly stable. This indicates that some crabs are likely able to stop (or at least drastically slow) the progress of the infection. And again, we really don't know a lot of the timeline for _Hematodinium_ infection. We also really can't extrapolate much from the pathology of _H. perezi_, which seems to have an extremely short time between infection and mortality. Lot of work should really be done here.

- When applying for grants, often people won't understand how knowledge about diseases can be used - they'll look at it and say why does this matter, we can't stop the disease from spreading. Focus a lot on the impact of the work to compensate - how knowledge about diseases helps management/improves models/etc.

#### Sub-conversation: Extracting Samples:

One subject we touched on was the process of collecting and extracting samples. This is pretty important, so I'm putting it in a separate subhead. More information on the specific methods used to collect can be found [here](https://github.com/afcoyle/hemat_bairdi_transcriptome/blob/main/data/jensen_archived_samples/jensen_data/Hemato_protocols/Hemato_DNA_extraction_protocol_062220.doc)

- Plates were pre-filled with EtOH, and hemolymph was injected right through the caps

- At this point, best-case scenario is that the samples have a sort of cottage-cheesy texture. Worst-case is that they're dry flakes. Both are workable, neither can be resuspended by just adding EtOH and mixing it around

- If cottage cheese, spin it down, pour off EtOH, and treat as a tissue sample. Flipping and pouring off the EtOH is nerve-wracking, but genuinely the best way to do it

- If doing the whole plate, don't try to replace the cap mat. You'll inevitably cause contamination, just replace altogether

- It might be best to practice messing around with the cap of the negative control well a few times, just to get a feel for it

- It's likely that all are dried out by now, but just in case, it'd likely be best to start on the 2019 samples - they're the most recent, so least likely to be dry

- Pam tried completely drying the plate and lysing with proteinase K. Never got it to all lyse, and it didn't really work well.

- For extraction, would take 150 uL of the cottage cheese. Each sample consisted of ~1000 uL total (before evaporation)

- Should likely play around with unimportant plates first

- **There might be a folder in one of the totes that contains a lot of important sample info** [unclear if this is the same as the notebook mentioned well above]

- If samples are still wet, put in the shaking incubator overnight to homogenize as much as possible

- Use the big pipette tips, and cut off the end of the tips to minimize clogging

- **Use filter tips!** Inevitably, the cottage cheese will clog the pipette tip, and then the unclogging will result in the sample being sprayed up. Without a filter tip, this means certain contamination.

- **Don't try to evaporate away the EtOH**, you'll just contaminate the lab. In fact, keep uncapped plates out of warm incubators to minimize evaporation