---
layout: post
title: Coral Gene Expression/DNA Methylation Lit Review
date: '2022-07-14'
categories: literature
tags: review
---

[Link](https://paperpile.com/shared/WjKdn6) to shared paperpile folder of literature review

### Overview

DNA methylation analysis has the capacity to inform our understanding of how environmental memory (parent to offspring) impacts phenotypic plasticity and responses to climate change. Invertebrates have sporadic methylation throughout their genomes (mosaic), unlike vertebrate taxa with more consistent and widespread methylation. Increased methylation can correlate with increased transcription. The impacts to expression caused by methylation can be thought of as both a long term and short term changes to the marine invertebrate genome, happening within the lifetime of an organism, but with the potential to be passed to offspring and cause CpG depletion within the genome over time. The inconsistent mosaic of methylation seems to generally point to the conclusion of DNA methylation having importance in only certain instances of expression within the genome. Whether or not those areas are dictated by environmental stimuli is up for debate. 

Differential expression/methylation analysis between different treatments or corals found in different areas appears to be the primary method of incorporating this environment piece. And while publications have discovered different patterns depending on environment/stressors in both methylation and expression using these methods, definitive understanding of how methylation impacts gene expression involved in environmental stress response limit their conclusions. This mechanistic piece seems more tied to understanding the roles a wide variety of genes play in stress response and whether or not researchers have a clear picture of which genes are more involved in different stress responses (heat, OA, etc.).

In the examination of relationships between expression and methylation, the gold standard is a whole genome approach, with the ability to compare whole genomes, transcriptomes, and methylomes (WGS, RNA-seq, WGBS). More targeted methods like RRBS and MBDBS are not comprehensive and do not include the high resolution single base pair data necessary for a full analysis. A targeted approach could work for a more specific research question, but our interests seem to fall into identification of all relationships that may exist between DNA methylation and expression. In this case, our analysis would require a full reference genome, RNA-seq representing the full transcriptome as fully as possible, and whole methylome data from WGBS.

When analyzing DNA methylation it is important to consider all of the potential reasons methylation may be present.It may reduce gene expression variability, spurious  expression, and silencing  of  repeated  genetic  elements–all of which are functions more related to gene expression maintenance rather than response to environmental stimuli. In the case of M. capitata and P. acuta, more methylation in M. capitata is associated with a higher tolerance to environmental variability, while less methylation in P. acuta is associated with more environmental sensitivity. However, M. capitata’s genome is over twice the size of P. acuta, so it is assumed that higher rates of DNA methylation are at least in part due to management of genome efficiency.

Common genes targeted for analysis in their potential relevance to environmental response involve cell signaling, heat-shock response, and those acting as molecular chaperones. Variability in DNA methylation of genes like this is predominantly limited to analysis of methylation data without paired gene expression data (see following section). Another theme that came up repeatedly is seasonal variation in environmental response, with DNA methylation patterns changing seasonally and adding another level of complexity to be controlled for or examined when it comes to relationships between the environment and methylation.

### Gaps to fill

I had difficulty finding papers that examined coupled DNA methylation and gene expression data at the same time, which makes sense why we would be trying to fill a gap like this since we have paired expression and methylation datasets from the Putnam Lab’s Express_Compare/Meth_Compare repos. But I think I probably missed something so my big circle back will be to look further into this for coral, but more than likely other species. Still not 100% how lncRNA could fit into this, but I assume it could be considered similar to DNA methylation in the same set of analyses given its influence on gene expression. So maybe it could look like an analysis of how DNA methylation impacts gene expression and how expressed lncRNAs could do the same. 
